packages/commonwealth/client/scripts/views/pages/Editor/Editor.tsx

```diff
@@ -0,0 +1,262 @@
+  const imageUploadHandlerDelegateLocal = useImageUploadHandlerLocal();
+  const imageUploadHandlerDelegateS3 = useImageUploadHandlerS3();
+
+  return useCallback(
+    async (file: File): Promise<ImageURL> => {
+      switch (imageHandler) {
+        case 'S3':
+          return await imageUploadHandlerDelegateS3(file);
+        case 'local':
+          return await imageUploadHandlerDelegateLocal(file);
+      }
+
+      throw new Error('Unknown image handler: ' + imageHandler);
+    },
+    [
+      imageHandler,
+      imageUploadHandlerDelegateLocal,
+      imageUploadHandlerDelegateS3,
+    ],
+  );
+}
+
+export const Editor = memo(function Editor(props: EditorProps) {
+  const errorHandler = useEditorErrorHandler();
+  const [dragging, setDragging] = useState(false);
+  const [uploading, setUploading] = useState(false);
+
+  const dragCounterRef = useRef(0);
+
+  const mode = props.mode ?? 'desktop';
+  const imageHandler: ImageHandler = props.imageHandler ?? 'S3';
+
+  const placeholder = props.placeholder ?? 'Share your thoughts...';
+
+  const mdxEditorRef = React.useRef<MDXEditorMethods>(null);
+
+  const imageUploadHandlerDelegate = useImageHandler(imageHandler);
+
+  const imageUploadHandler = useCallback(
+    async (file: File) => {
+      try {
+        setUploading(true);
+        return await imageUploadHandlerDelegate(file);
+      } finally {
+        setUploading(false);
+      }
+    },
+    [imageUploadHandlerDelegate],
+  );
+
+  const handleFile = useCallback(async (file: File) => {
+    const text = await fileToText(file);
+    mdxEditorRef.current?.setMarkdown(text);
+  }, []);
+
+  const handleImportMarkdown = useCallback(
+    (file: File) => {
+      async function doAsync() {
+        await handleFile(file);
+      }
+
+      doAsync().catch(console.error);
+    },
+    [handleFile],
+  );
+
+  const handleDropAsync = useCallback(
+    async (event: React.DragEvent) => {
+      console.log(event.dataTransfer.files.length);
+
+      const nrFiles = event.dataTransfer.files.length;
+
+      try {
+        if (nrFiles === 1) {
+          const type = event.dataTransfer.files[0].type;
+
+          if (['text/markdown', 'text/plain'].includes(type)) {
+            await handleFile(event.dataTransfer.files[0]);
+          } else {
+            // TODO: use a snackbar
+            console.log('File not markdown');
+          }
+        }
+
+        if (nrFiles <= 0) {
+          // TODO: use a snackbar
+          console.log('No files given');
+          return;
+        }
+
+        if (nrFiles > 1) {
+          // TODO: use a snackbar
+          console.log('Too many files given');
+          return;
+        }
+      } finally {
+        setDragging(false);
+      }
+    },
+    [handleFile],
+  );
+
+  // TODO: handle html files but I'm not sure about the correct way to handle it
+  // because I have to convert to markdown.  This isn't really a typical use
+  // case though
+  const handleDrop = useCallback(
```

Comment:
Bug: dropping a markdown files says "Unsupported file"

Reply:
Can you elaborate on this? I don't understand.

Reply:
Basically dropping any markdown file in the editor drop window doesn't work.



https://github.com/user-attachments/assets/a0039a2c-3500-4e03-8707-941294a599e7

Reply:
Could you try again? Or we could wait until v1... The issue is that the media type is being set incorrect.

I updated it so it will print the media type so we can at least fix it at that point.  Just need to figure it why the type is wrong.

Reply:
v1 will do, thanks for looking.